src/main/java/org/tinyradius/packet/RadiusPacket.java

```diff
@@ -1007,7 +1018,7 @@ protected static RadiusPacket decodePacket(Dictionary dictionary, InputStream in
 			throw new RadiusException("bad packet: attribute length mismatch");
 
 		// create RadiusPacket object; set properties
-		RadiusPacket rp = createRadiusPacket(type);
+		RadiusPacket rp = createRadiusPacket((forceType == UNDEFINED) ? type : forceType);
```

Comment:
The caller cannot set the value because when the packet received is authentication(AccessRequest), if secret does not match the correct one it returns an exception. 

The forcetype is to make a simple packet without trying to decode anything using secret , when trying to figure out what is the secret we dont have the correct secret to decode the packet authentication data.